finality-provider/service/errors.go

```diff
@@ -1,6 +1,22 @@
 package service
 
-import "errors"
+import (
+	"errors"
+	"fmt"
+
+	bbntypes "github.com/babylonlabs-io/babylon/types"
+)
+
+const instanceTerminatingMsg = "terminating the finality-provider instance due to critical error"
```

Comment:
Why is this not a type of `CriticalError`?

Reply:
Did you mean why the message not part of `CriticalError`? This is because this is only used in https://github.com/babylonlabs-io/finality-provider/blob/64ae9cacff4d91919391cc99c6c630d3649f06a9/finality-provider/service/event_loops.go#L177 where we decide to terminate the instance. Not for all of the cases